packages/xo-server-sdn-controller/src/index.js

```diff
@@ -478,6 +491,23 @@ class SDNController extends EventEmitter {
               )
             }
 
+            // Previously created network didn't store `vlan`
+            //
+            // 2020-01-27
+            // This is used to add the `vlan` to networks created before this version. (v0.4.0)
+            // This will be removed in 1 year.
+            if (otherConfig['xo:sdn-controller:vlan'] === undefined) {
```

Comment:
No, the above block is entered only if `otherConfig['xo:sdn-controller:pif-device']` is `undefined` which is not necessarilly the case.

Reply:
I just did this comment because `getHostTunnelForNetwork` will be called two times in case of not-existent `pif-device` and not-existent `xo:sdn-controller:vlan` in the same time and two avoid logic duplication.

Reply:
I know but I need it to be called if only non-exsitent `pif-device` and only non existent `vlan`.

And the case where both are non exsistent should be very rare. This is comaptibility code or different versions taht will be removed in the coming year.